### PR TITLE
LibJS: Add Function() and Function.prototype

### PR DESCRIPTION
**Pull request by linusg**
This adds:

- `[new] Function()`
  - https://www.ecma-international.org/ecma-262/5.1/#sec-15.3.2.1
  - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/Function
- `Function.prototype.apply()`
  - https://www.ecma-international.org/ecma-262/5.1/#sec-15.3.4.3
  - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/apply
- `Function.prototype.call()`
  - https://www.ecma-international.org/ecma-262/5.1/#sec-15.3.4.4
  - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/call
- `Function.prototype.toString()`
  - https://www.ecma-international.org/ecma-262/5.1/#sec-15.3.4.2
  - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/toString

I also added a dummy for `Function.prototype.bind()`, but implementing that will require some more work - probably either extending `ScriptFunction` or adding a separate class for bound functions (https://www.ecma-international.org/ecma-262/5.1/#sec-15.3.4.5).

I know that there are a few FIXMEs in this patch, but those seem out of scope for this PR :^)
(also I want to not make it any larger)

### CODE REVIEW

File: Libraries/LibJS/Runtime/FunctionConstructor.cpp [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        for (size_t i = 0; i < interpreter.argument_count() - 1; ++i)
+            parameters.append(interpreter.argument(i).to_string());
+        StringBuilder parameters_builder;
+        parameters_builder.join(',', parameters);
+        parameters_source = parameters_builder.build();
+        body_source = interpreter.argument(interpreter.argument_count() - 1).to_string();
+    }
+    auto source = String::format("function (%s) { %s }", parameters_source.characters(), body_source.characters());
+    auto parser = Parser(Lexer(source));
+    auto function_expression = parser.parse_function_node<FunctionExpression>();
```
Comment (linusg) [inline Libraries/LibJS/Runtime/FunctionConstructor.cpp:69]: This seemed like the most straightforward way without having to add something like `Parser::parse_function_parameters()` and `Parser::parse_function_body_without_curly_braces()` :^)

Comment (awesomekling) [reply Libraries/LibJS/Runtime/FunctionConstructor.cpp:69]: Yeah it's okay for now :)

File: Libraries/LibJS/Runtime/FunctionPrototype.cpp [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+FunctionPrototype::FunctionPrototype()
+{
+    put_native_function("apply", apply);
+    put_native_function("bind", bind);
+    put_native_function("call", call);
+    put_native_function("toString", to_string);
+    put("length", Value(0));
+    get("apply").value().as_object().put("length", Value(2));
+    get("bind").value().as_object().put("length", Value(1));
+    get("call").value().as_object().put("length", Value(1));
```
Comment (awesomekling) [inline Libraries/LibJS/Runtime/FunctionPrototype.cpp:47]: This seems like it should happen inside `put_native_function` instead. :)

Comment (linusg) [reply Libraries/LibJS/Runtime/FunctionPrototype.cpp]: Agreed, this is weird. Updated :)

**Comment by linusg** [comment]
@awesomekling I've added the length property for `NativeFunction`s and `ScriptFunction`s in two separate commits.

**Comment by awesomekling** [comment]
Very good!

**Pull request merged**